actix-server/src/accept.rs

```diff
@@ -97,13 +51,23 @@ fn connection_error(e: &io::Error) -> bool {
 }
 
 impl Accept {
-    pub(crate) fn start(
-        poll: Poll,
-        waker: WakerQueue,
-        socks: Vec<(Token, MioListener)>,
-        srv: Server,
-        handles: Vec<WorkerHandle>,
-    ) {
+    pub(crate) fn start<F>(
+        sockets: Vec<(Token, MioListener)>,
+        server: Server,
+        worker_factory: F,
+    ) -> WakerQueue
+    where
+        F: FnOnce(&WakerQueue) -> Vec<WorkerHandle>,
```

Comment:
the thing is server builder and accept both want a copy of worker handles and you need a waker_queue to build a worker.
a fn once is just convinent to let server builder can collect a copy of handles easier.

Reply:
This can be worked out later when rtless actually lands. As the only reason accept keeps a copy of waker_queue is to wake itself up after a timeout future.